tickets/DM-32723 (#7)
lsst-dmsst/DM_SP_study: sec_proc.tex

\section{Details and Challenges of Processing Data from Special Programs}\label{sec:proc}

A discussion of the anticipated details and challenges related to 
obtaining and processing data from Special Programs.

Appendix~\ref{sec:spcs} provides detailed examples for the processing
of data from Special Programs, including scenarios in which regular,
Special, and User-Generated Processing are all involved.

\subsection{Boundaries on Non-Standard Visits} \label{ssec:proc_bounds}

Special Programs that do not use standard or alternative standard visits
might be affected by hardware or processing boundaries.

\subsubsection{Hardware Boundaries}\label{ssec:proc_bounds_hardware}

Appendix~\ref{sec:hardbounds} lists all of the hardware boundaries that 
might constrain the potential diversity of Special Programs data.

In general, the currently-proposed Special Programs in \citeds{PSTN-055}
are not anticipated to be limited by hardware boundaries.

A few potential challenges posed by hardware
boundaries are summarized below.

\begin{itemize}

\item \textbf{Short exposures --}
Special Programs that use short exposures would be limited to the

> minimum exposure time is 1 second (stretch goal: 0.1 seconds\footnote{A short-exposure survey 

Reviewer: ```suggestion
minimum exposure time of 1 second (stretch goal: 0.1 seconds\footnote{A short-exposure survey 
```

of the bright stars of M67, described in Chapter 10.4 of the 
Observing Strategy White Paper \citep{2017arXiv170804058L}, suggests using the stretch goal of 
0.1 second exposures or, if that is not possible, \textit{"custom pixel masks to accurately perform 
photometry on stars as much as 6 magnitudes brighter than the saturation level"}. 
This may require User-Generated Processing.}).
There is a potential hardware boundary that limits the readout rate to 1 
every 15 seconds, which would affect the image acquisition rate and 
increase the overheads on short exposures.

\item \textbf{Repeated pointing --}

> Special Programs which require the \emph{exact same} field pointing and 

Reviewer: ```suggestion
Special Programs that require the \emph{exact same} field pointing and 
```

Reviewer: (restrictive clause)

rotation for \emph{every exposure} (to sub-arcsecond levels) might run 
into hardware boundaries on pointing and tracking.

\item \textbf{Twilight images --}
Special Programs that obtain twilight images will be subject to safe
limits on sky background flux, as with any astronomical camera.

\end{itemize}

Finally, as a side note, Special Programs that request a high number of 
filter changes and/or long slews could be inefficient due to large overheads,
but would not be limited by hardware boundaries.

\subsubsection{Processing Boundaries}\label{ssec:proc_bounds_processing}

Appendix~\ref{sec:procbounds} describes the boundaries on what types of visits 
can be processed and calibrated by the Data Management System and the LSST
Science Pipelines.

Most of the currently-proposed Special Programs in \citeds{PSTN-055}
are not anticipated to be limited by hardware boundaries.
However, those which use non-standard visits, especially those with
short exposure time or those obtained during twilight, might
be affected by processing boundaries.

The most likely challenges posed by processing
boundaries are summarized below.

\begin{itemize}

\item \textbf{Very short exposures --}

> Special Programs which use very short ($<$2 sec) exposures 

Reviewer: ```suggestion
Special Programs that use very short ($<$2 sec) exposures 
```

Reviewer: (restrictive clause)

could be difficult to reduce due to an incompletely-formed PSF 
(Section~\ref{ssec:procbounds_expt}).
The Data Management System is required to be able to process exposure 
times as low as 1 second (Section~\ref{ssec:req_proc}), 
but it is known that such short exposures might have degraded image quality.

\item \textbf{Very short or very long exposures --}
Special Programs that use very short or very long ($>$150 sec) 
exposures could be difficult to calibrate due to having too few 
(or too few unsaturated) stars.

\item \textbf{Twilight images --}

> For Special Programs which obtain images with very bright sky backgrounds

Reviewer: ```suggestion
For Special Programs that obtain images with very bright sky backgrounds
```

(twilight images), it is currently unclear whether they can be processed
with the LSST Science Pipelines; 
User-Generated Processing might be needed (e.g., \citealt{2022AJ....164..168S}).

\item \textbf{Streaked images --}
The full reduction and calibration of data from any Special Programs that 
use non-sidereal tracking, which produce images with star streaks, is
currently beyond the scope of the LSST Science Pipelines; 
User-Generated Processing would be needed.

\end{itemize}

\subsection{Regular (non-special) processing}\label{ssec:proc_reg}

Recall that the term ``regular processing" is used only within this document,
and refers to the image processing pipelines that are designed for, 
and will be applied to, the Main Survey's observations (Section~\ref{sec:intro}).

Decisions about when to apply regular processing to Special Programs data,
or when to include it in the data products for the Main Survey
(e.g., if it improves the all-sky coadds), 
are ultimately left to the discretion of the Rubin Observatory's 

> Data Management and System Performance teams.

Reviewer: ```suggestion
Data Management and System Performance departments in Rubin Operations.
```

then unique and separate data products could be generated with Special Processing.
For the NEO twilight survey, these data products would likely be similar to the
Prompt or DIA data products.
For the calibration survey, these data products would like be similar to the
annual data release tessellated coadds and associated catalog.

\subsubsection{Standard Visit Mini-Surveys}

Consider a Special Program in which a special region of sky is observed with standard
visits but a a special strategy or cadence which is significantly distinct from the Main Survey
observing modes, and lasts for a limited amount of time.
For example, a short-term survey of the Magellanic Clouds, with dual primary science
goals in time-domain and static-sky science.

In cases like this, a set of unique and separate data products with the same formats as the 
time-domain DIA and static-sky data products described in Section~\ref{sssec:proc_reg_dr}
should be created with Special Processing.
They might be released with an annual data release or on an intermediate timescale, e.g.,
within six months of the conclusion of the mini-survey observations.

\subsubsection{Target-of-Opportunity (TOO) Observations}

> Options for RDM to process TOO observations, especially during the first year of Operations 

Reviewer: ```suggestion
Options for Data Management to process TOO observations, especially during the first year of Operations 
```
